From 6d193611bb9baee7b044cdc3a8351f2c40bca94d Mon Sep 17 00:00:00 2001
From: Rafal Krypa <r.krypa@samsung.com>
Date: Wed, 19 Nov 2014 10:54:31 +0100
Subject: [PATCH 21/24] Add inheritable CAP_MAC_OVERRIDE to user session

User sessions already have inheritable CAP_MAC_ADMIN permission for setting
Smack labels in application launcher. It seems that also CAP_MAC_OVERRIDE
will be needed to bypass Smack checks during modification of socket labels:

kernel: type=1400 audit(1416390778.371:3): lsm=SMACK fn=smack_inode_setxattr
action=denied subject="User" object="_" requested=w pid=2422
comm="amd_session_age" name="UNIX" dev="sockfs" ino=14108

CAP_MAC_ADMIN only allows policy administration, it doesn't override the
checks.

Change-Id: I7a2dbae2391989d93a0858925d5b15c1ef9a880c
Signed-off-by: Rafal Krypa <r.krypa@samsung.com>
---
 units/user@.service.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/units/user@.service.in b/units/user@.service.in
index 2e25feb..def6a1f 100644
--- a/units/user@.service.in
+++ b/units/user@.service.in
@@ -18,5 +18,5 @@ SmackExecLabel=User
 Slice=user-%i.slice
 KillMode=mixed
 Environment=DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/%U/dbus/user_bus_socket
-Capabilities=cap_mac_admin,cap_setgid=i
+Capabilities=cap_mac_admin,cap_mac_override,cap_setgid=i
 SecureBits=keep-caps
-- 
1.8.4.5

